servers/quarkus-cli/src/main/java/org/projectnessie/quarkus/cli/CheckContent.java

```diff
@@ -0,0 +1,204 @@
+    warnOnInMemory();
+
+    StoreWorker<?, ?, ?> worker =
+        (StoreWorker<?, ?, ?>)
+            getClass()
+                .getClassLoader()
+                .loadClass(workerClass)
+                .getDeclaredConstructor()
+                .newInstance();
+
+    Hash hash = hash();
+    if (contentKey != null && !contentKey.isEmpty()) {
+      check(hash, List.of(Key.of(contentKey)), worker);
+    } else {
+      List<Key> batch = new ArrayList<>(batchSize);
+      try (Stream<KeyListEntry> keys = databaseAdapter.keys(hash, KeyFilterPredicate.ALLOW_ALL)) {
+        keys.forEach(
+            keyListEntry -> {
+              batch.add(keyListEntry.getKey());
+
+              if (batch.size() >= batchSize) {
+                check(hash, batch, worker);
+                batch.clear();
+              }
+            });
+
+        check(hash, batch, worker); // check remaining keys
+      }
+    }
+
+    return errorDetected.get() == 0 ? 0 : 2;
+  }
+
+  private Hash hash() throws ReferenceNotFoundException {
+    if (hash != null) {
+      return Hash.of(hash);
+    }
+
+    String effectiveRef = ref;
+    if (effectiveRef == null) {
+      effectiveRef = serverConfig.getDefaultBranch();
+    }
+
+    ReferenceInfo<ByteString> main =
+        databaseAdapter.namedRef(effectiveRef, GetNamedRefsParams.DEFAULT);
+    return main.getHash();
+  }
+
+  private void check(Hash hash, List<Key> keys, StoreWorker<?, ?, ?> worker) {
+    Map<Key, ContentAndState<ByteString>> values;
+    try {
+      values = databaseAdapter.values(hash, keys, KeyFilterPredicate.ALLOW_ALL);
+    } catch (Exception e) {
+      keys.forEach(k -> report(k, e));
+      return;
+    }
+
+    keys.forEach(
+        k -> {
+          if (values.get(k) == null) {
+            report(k, "Missing content", null);
+          }
+        });
+
+    values.forEach(
+        (k, contentAndState) -> {
+          try {
+            Object value =
+                worker.valueFromStore(
+                    contentAndState.getRefState(), contentAndState::getGlobalState);
+            report(k, null, value);
+          } catch (Exception e) {
+            report(k, e);
+          }
+        });
+  }
+
+  private void report(Key key, Throwable error) {
+    if (verbose) {
+      spec.commandLine().getErr().printf("Error for key %s%n", key);
+      error.printStackTrace(spec.commandLine().getErr());
+    }
+
+    StringBuilder msg = new StringBuilder(error.toString());
+    while ((error = error.getCause()) != null) {
+      msg.append("; Caused by: ");
+      msg.append(error);
+    }
+
+    report(key, msg.toString(), null);
+  }
+
+  private void report(Key key, String error, Object content) {
+    if (error != null) {
+      errorDetected.incrementAndGet();
+    }
+
+    if (error == null && errorOnly) {
+      return;
+    }
+
+    PrintWriter out = spec.commandLine().getOut();
+    out.printf(error == null ? "OK" : "ERROR");
+    out.print(separator);
+    out.print(key);
```

Comment:
Instead of just `key` maybe `UriUtil.toPathString(key.getElements())`

Reply:
good point.

Reply:
I guess encoding is not required now that the output is JSON.